justfile

```diff
@@ -0,0 +1,42 @@
+set dotenv-load
+set ignore-comments
+
+# Default recipe
+[private]
+default:
+    @just --list
+
+# Install necessary tools
+install-tools:
+    rustup component add rustfmt --toolchain nightly
+    cargo install cargo-sort
+
+# Check if necessary tools are installed
+[private]
+check-tools:
+    #!/usr/bin/env bash
+    if ! cargo +nightly fmt --help &> /dev/null; then 
+        echo "cargo +nightly fmt is not installed, use just install-tools or install it manually"
+        exit 1
+    fi
+    if ! cargo sort --help &> /dev/null; then 
+        echo "cargo sort is not installed, use just install-tools or install it manually"
+        exit 1
+    fi
+
+# Format code
+format: check-tools
+    cargo clippy --all-targets --all-features
+    cargo +nightly fmt -- --check
+    cargo sort -w . --check
+
+# Build all binaries and debian packages
+build *ARGS:
+    ci/build_scripts/build.sh {{ARGS}}
+
+# Run unit tests
+test: format
+    cargo test --no-fail-fast --all-features
+
+# Run integration tests
+test-integration:
```

Comment:
That was a part when @reubenmiller would add integration tests, but we decided to move into another PR, so now it is deleted.

Reply:
> That was a part when @reubenmiller would add integration tests, but we decided to move into another PR, so now it is deleted.

I will add the integration tests in a follow up PR (as I still need to look how it will work).